LandSandBoat/server
# Improves !pos command by handling commas

**_I affirm:_**

## What does this pull request do?

* Improves `!pos` command to make it more usable for developers:

Formats now handled:
* `!pos 1.0 2.0 3.0`
* `!pos 1.0, 2.0, 3.0`
* `!pos 1.0, 2.0, 3.0,`
* `!pos 1.0,2.0,3.0`
* `!pos 1.0,2.0,3.0,`
* `!pos 1 2 3`
* `!pos 1, 2, 3`
* `!pos 1, 2, 3,`
* `!pos 1,2,3`
* `!pos 1,2,3,`

You can now easily paste positions from SQL or Lua, without needing to manually remove commas.

* Adds zone ID to the default `!pos`, when displaying current position.
* Changes message to use `xi.msg.channel.SYSTEM_3` for much more readable text.


## Steps to test these changes
Run `!pos` again to verify the new position is correct, or append print after setting position eg.
```lua
zoneId = targ:getZoneID()
targ:setPos(x, y, z, 0)
player:PrintToPlayer(string.format("%s's position: X %.4f  Y %.4f  Z %.4f  Rot %i  (Zone: %i)", targ:getName(), targ:getXPos(), targ:getYPos(), targ:getZPos(), targ:getRotPos(), targ:getZoneID()), xi.msg.channel.SYSTEM_3)
```
